src/audits/performance/input-readiness-metric.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Audit = require('../audit');
+
+/* global window */
+window.global = window;
+
+// we need gl-matrix and jszip for traceviewer
```

Comment:
can wait to integrate this with `tracing-processor.js` in a followup PR, but can you still pull this into a separate file (and require it here) so e.g. fixing fCP scoring isn't blocked while waiting for the followup changes?